### ISSUE TITLE
Using loading.tsx with [...catchall] route in the parallel route breaks the rendering completely

### ISSUE THREAD

**Issue by max-mayorov** [vercel/next.js#60682]
### Link to the code that reproduces this issue

https://github.com/max-mayorov/nextgram-loading-catchall-bug

### To Reproduce

1. Start the application in development: `pnpm dev`
2. Click on any nav link on the top of the page:
![image](https://github.com/vercel/next.js/assets/4355960/6a7bb2a0-7aea-413a-9756-1dde83402423)

### Current vs. Expected behavior

Expected: 
About page opens, showing navigation buttons and page title (About, About Us, About Me)

Actual:
Nothing is loaded, white page

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Pro
Binaries:
  Node: 18.18.2
  npm: N/A
  Yarn: N/A
  pnpm: 8.14.0
Relevant Packages:
  next: 14.0.5-canary.58
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local), Vercel (Deployed), Other (Deployed)

### Additional context

This incorrect behavior is related to the combination of 
* `[...catchAll]` route inside parallel route `@modal` to dismiss the modal
* `loading.tsx` file inside affected pages. 

Removing `loading.tsx` _or_ `[...catchAll]` route can be a workaround fix the problem but in our project we need both of them unfortunately

**Comment by icyJoseph** [comment]
Using a Suspense boundary around the data fetching areas, instead of a loading.tsx file, seems to do the trick. I'll test a bit more.

**Comment by ztanner** [comment]
Hi! Thanks for the report. I checked your code reproduction and confirmed it's been fixed by #61597. Let us know if you are still encountering any problems after upgrading!
